addons/im_livechat/static/src/views/discuss_channel_list/discuss_channel_list_view_controller.js

```diff
@@ -1,20 +1,35 @@
 /* @odoo-module */
 
-import { _t } from "@web/core/l10n/translation";
+import { useState } from "@odoo/owl";
+
 import { ListController } from "@web/views/list/list_controller";
+import { useService } from "@web/core/utils/hooks";
+import { _t } from "@web/core/l10n/translation";
 
 export class DiscussChannelListController extends ListController {
     setup() {
         super.setup(...arguments);
+        this.threadService = useService("mail.thread");
+        this.store = useState(useService("mail.store"));
+        this.ui = useState(useService("ui"));
     }
 
     async openRecord(record) {
-        if (!record.data.is_member) {
-            return super.openRecord(record);
+        if (!this.ui.isSmall) {
+            return this.actionService.doAction("mail.action_discuss", {
+                name: _t("Discuss"),
+                additionalContext: { active_id: record.resId },
+            });
         }
-        this.actionService.doAction("mail.action_discuss", {
-            name: _t("Discuss"),
-            additionalContext: { active_id: record.resId },
+        let thread = this.store.Thread.get({
+            model: "discuss.channel",
+            id: record.resId,
         });
+        if (!thread) {
+            thread = await this.threadService.fetchChannel(record.resId);
+        }
+        if (thread) {
+            this.threadService.open(thread);
+        }
```

Comment:
The call to super cannot be removed, the function should still be able to open different kind of records (from mailboxes). Specific code should only be for channel.